### DISCUSSION TITLE
Locust master and worker authentication

### DISCUSSION THREAD

**Discussion by brvaland** [locustio/locust#2928] [General]
I am newbie to load test tools and exploring automation where we can trigger load test using locust via git action.

I would like to check if there is way to authenticate worker with master using password or certificate if provided.

Master and worker will run on different self hosted runner so need to ensure additional security on top of port binding.

**Comment by cyberw** [comment]
Hi! There's no such support at the moment, and the zeromq connection is not encrypted so any password would have to be sent in plain text, unless we really reworked a lot of stuff. 

Our hosted [locust.cloud](https://locust.cloud) solution has security built in and supports github action runs though!

**Reply by brvaland** [reply]
Thanks @cyberw for sharing details about locust.cloud.

With reference to sending password from worker to master over zeromq when connected. 

Is it possible for you share code snippets how we can achieve?

Can we write python code password validation on master / worker ?

**Reply by cyberw** [reply]
On locust.cloud, security is achieved by having every locust master/worker cluster on an isolated virtual network, so there's no code to share I'm afraid.
